packages/generator/src/parser/parser-types-v2.ts

```diff
@@ -0,0 +1,69 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+import {
+  EdmxMetadataBase,
+  EdmxEntityTypeBase,
+  EdmxFunctionImportBase,
+  EdmxEntitySetBase,
+  EdmxComplexTypeBase
+} from './parser-types-common';
+
+export interface EdmxMetadata extends EdmxMetadataBase {
+  entitySets: EdmxEntitySet[];
+  entityTypes: EdmxEntityType[];
+  associationSets: EdmxAssociationSet[];
+  associations: EdmxAssociation[];
+  functionImports: EdmxFunctionImport[];
+  complexTypes: EdmxComplexTypeBase[];
+}
+
+export function isV2Metadata(
+  metadata: EdmxMetadataBase
+): metadata is EdmxMetadata {
```

Comment:
is a boolean returned? what is metadata is EdmxMetadata for a type?

Reply:
This is called a type guard in Typescript (https://www.typescriptlang.org/docs/handbook/advanced-types.html#typeof-type-guards)